pinot-common/src/main/java/org/apache/pinot/common/config/provider/TableCache.java

```diff
@@ -57,10 +65,13 @@
   private static final String LOWER_CASE_OFFLINE_TABLE_SUFFIX = "_offline";
   private static final String LOWER_CASE_REALTIME_TABLE_SUFFIX = "_realtime";
 
+  private final Set<TableConfigChangeListener> _tableConfigChangeListeners = new HashSet<>();
```

Comment:
It'd be good to use `ConcurrentHashMap.newKeySet()` here to align with `_tableConfigMap`.